flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -530,6 +532,10 @@ protected Counter setupNumRecordsInCounter(StreamOperator streamOperator) {
 
     @Override
     public void restore() throws Exception {
+        runWithCleanUpOnFail(this::executeRestore);
```

Comment:
Would it be better to make `restore` also `final`, and let the subclasses to override `executeRestore` instead ? Since the exception handling should be always required